tools/run_backups

```diff
@@ -32,10 +32,65 @@ echo $(date +%T) : Backing up Bastillion-jetty directory ...
 #echo $(date +%T) : I would delete $(ls -1 bastillion/bastillion.*.gz | grep -v 20....01.tar.gz | head -n -3)
 
-echo $(date +%T) : Backing up TRSS using mongodump ...
-ssh backups@54.78.186.5 mongodump --archive --db=exampleDb | xz > trss/trss.$(date +%Y%m%d).tar.xz
+# echo $(date +%T) : Backing up TRSS using mongodump ...
+# ssh backups@54.78.186.5 mongodump --archive --db=exampleDb | xz > trss/trss.$(date +%Y%m%d).tar.xz
 #echo $(date +%T) : I would delete $(ls -1 trss/trss.*.?z | grep -v 20....01.tar..z | head -n -3)
-rm -v $(ls -1 trss/trss.*.?z | grep -v 20....01.tar..z | head -n -3)
+# rm -v $(ls -1 trss/trss.*.?z | grep -v 20....01.tar..z | head -n -3)
+# Improved TRSS Backup Script, Daily Incremental Of Key Collections
+# Weekly Full Backup On Sunday
+# TRSS Restart Script Scheduled On TRSS @ 5am Sunday
+
+echo ""$(date +%T)" : Starting TRSS backup with remote lock..."
+TRSS_REMOTE_HOST="backups@20.90.182.165"
+LOCKFILE="/tmp/trss-backup.lock"
+
+# Check for existing lock
+if ssh "$TRSS_REMOTE_HOST" "[ -f $LOCKFILE ]"; then
+  echo ""$(date +%T)" : Remote backup lock already present, exiting..."
+  exit 1
+fi
+
+# Set remote lock
+ssh "$TRSS_REMOTE_HOST" "touch $LOCKFILE"
+trap 'ssh "$TRSS_REMOTE_HOST" "rm -f $LOCKFILE"' EXIT
+
+echo "$(date +%T)" : Lock acquired at $LOCKFILE
+
+DOW=$(date +%w)  # 0 = Sunday
+
+if [ "$DOW" -eq 0 ]; then
+  echo ""$(date +%T)" : Performing full weekly backup of TRSS..."
+  ssh "$TRSS_REMOTE_HOST" docker exec aqa-test-tools-mongo-1 \
+    mongodump --archive --db=exampleDb \
+  | xz > trss/trss."$(date +%Y%m%d)".xz
+
+  echo ""$(date +%T)" : Cleaning up old full backups (keeping last 4)..."
+  FULL_BACKUPS=$(find trss -maxdepth 1 -name 'trss.[0-9]*.xz' | sort)
+  FULL_COUNT=$(echo "$FULL_BACKUPS" | wc -l)
+  if [ "$FULL_COUNT" -gt 4 ]; then
+    echo "$FULL_BACKUPS" | head -n $(($FULL_COUNT - 4)) | xargs -r rm -v
+  fi
+else
+  LAST_SUNDAY=$(date -d last-sunday +%Y-%m-%dT00:00:00Z)
+  echo "$(date +%T)" : Performing simulated daily incremental TRSS backup since $LAST_SUNDAY...
+
+  for collection in ApplicationTests auditLogs testResults; do
+    echo ""$(date +%T)" : Dumping updated documents from $collection..."
+    ssh "$TRSS_REMOTE_HOST" docker exec aqa-test-tools-mongo-1 bash -c \
+    "mongodump --db=exampleDb --collection=$collection \
+     --query='{ \"timestamp\": { \"\$gte\": { \"\$date\": \"$LAST_SUNDAY\" } } }' \
+     --archive --gzip" \
+    | xz > trss/trss.incr.${collection}."$(date +%Y%m%d)".archive.xz
+
+    echo ""$(date +%T)" : Cleaning up old incremental backups for $collection (keeping last 7)..."
+    INCR_BACKUPS=$(find trss -maxdepth 1 -name "trss.incr.${collection}.*.archive.xz" | sort)
+    INCR_COUNT=$(echo "$INCR_BACKUPS" | wc -l)
+    if [ "$INCR_COUNT" -gt 7 ]; then
+      echo "$INCR_BACKUPS" | head -n $(($INCR_COUNT - 7)) | xargs -r rm -v
+    fi
+  done
+fi
+echo "$(date +%T)" : TRSS backup complete. Remote lock released..
```

Comment:
Where is the lockfile released?

Reply:
This line , removes the lockfile, on any exit event.

trap 'ssh "$TRSS\_REMOTE\_HOST" "rm -f $LOCKFILE"' EXIT

Reply:
OK so it would hold it after any additional backups in the file after the TRSS block until they completed.

Reply:
I've put a fix in to make the lockfile and trap a little more robust.. so even if the run\_backups scripts aborts, it removes the lockfile, the restart script on trss should then run as normal.